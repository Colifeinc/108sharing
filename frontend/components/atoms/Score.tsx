import * as React from 'react'
import { ReviewStars } from 'components/molecules'
import classNames from 'classnames'

type Size = 'small' | 'medium' | 'large'

interface IProps {
  score: number
  size: Size
  omit?: boolean
  totalCount?: number
  textSize?: Size
  className?: string
}

const getSize = (size: Size) => {
  switch (size) {
    case 'small':
      return { iconWidth: 13, iconHeight: 12, gapMarginClassName: 'mx-2px' }
    case 'medium':
      return { iconWidth: 17, iconHeight: 16, gapMarginClassName: 'mx-2px' }
    default:
      return {}
  }
}

const getTextSize = (size: Size) => {
  switch (size) {
    case 'small':
      return 'text-sm ml-1'
    case 'medium':
      return 'ml-1'
    default:
      return 'text-lg ml-2'
  }
}

const Score: React.FC<IProps> = props => {
  const sizeProps = getSize(props.size ?? 'small')
  return (
    <div className={props.className}>
      <div className="flex items-center">
        {props.omit || props.score === 0 ? (
          <ReviewStars score={1} range={[1]} {...sizeProps} />
        ) : (
          <ReviewStars score={props.score} {...sizeProps} />
        )}
        <span
          className={classNames([
            'leading-none',
            getTextSize(props.textSize ?? 'small'),
            props.textSize != null && props.size === 'large' ? 'ml-2' : 'ml-1',
          ])}
        >
          <span className="font-bold">{props.score === 0 ? '-' : props.score}</span>
          {!!props.totalCount && props.totalCount !== 0 && (
            <span className="pl-2px">({props.totalCount})</span>
          )}
        </span>
      </div>
    </div>
  )
}

export default Score
