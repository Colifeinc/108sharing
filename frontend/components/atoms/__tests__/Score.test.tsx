import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import * as React from 'react'
import Score from '../Score'

jest.mock('../../../core/i18n')

describe('atoms/Score.tsx', () => {
  it('should matches to the snapshot', () => {
    const wrapper = shallow(<Score size="small" score={3.6} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
