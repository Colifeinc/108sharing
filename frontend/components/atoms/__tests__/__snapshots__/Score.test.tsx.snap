// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`atoms/Score.tsx should matches to the snapshot 1`] = `
<div>
  <div
    className="flex items-center"
  >
    <ReviewStars
      gapMarginClassName="mx-2px"
      iconHeight={12}
      iconWidth={13}
      score={3.6}
    />
    <span
      className="leading-none text-sm ml-1 ml-1"
    >
      <span
        className="font-bold"
      >
        3.6
      </span>
    </span>
  </div>
</div>
`;
